app/views/users/show.html.erb

```diff
@@ -4,62 +4,61 @@
-        <% end %>
+      <div class="panel panel-body">
+        <%= render 'public_user_data' %><br>
+        <%= render 'private_user_data' if @user == current_user %>
       </div>
     </div>
-<% end %>
 
-<% if @user == current_user %>
-    <div class="panel panel-default">
-      <div class="panel-heading">
-        <h4><%= t('.time_sheets_overview')%></h4>
-      </div>
-      <div class="panel-body">
-        <% @user.years_and_months_of_existence.each do |month| %>
-            <p>
-              <%= link_to l(Date.new(month.first, month.last, 1), format: :without_day) + ':', work_days_path(month: month.last, year: month.first) %>
-              <% current_user.projects_for_month(month.first, month.last).each do |project| %>
-                  <%= link_to project.title + " ", work_days_path(month: month.last, year: month.first, project: project.id, user_id: current_user) %>
-              <% end %>
-            </p>
-        <% end %>
-      </div>
-    </div>
+    <% if @user == current_user %>
+        <div class="panel panel-default">
+          <div class="panel-heading">
+            <h4><%= t('.signature') %></h4>
+          </div>
+          <div class="panel-body">
+            <% if current_user.signature.nil? %>
+                <b><%= t('.warning') %></b><br/>
+                <%= t('.information') %>
+                <p><%= form_tag({action: :upload_signature}, multipart: true) do %>
+                      <%= file_field 'upload', 'datafile' %><br/>
+                      <%= submit_tag t('.upload') %>
+                  <% end %></p>
+            <% else %>
+                <% path = 'data:image/png;base64,' << current_user.signature %>
+                <p><img src="<%= path %>" width="150" height="50"/></p><br/>
+                <%= link_to I18n.t('helpers.titles.delete', model: t('.signature')), delete_signature_path(current_user), method: :post, data: {confirm: t('helpers.links.confirm')}, class: 'btn btn-danger btn-xs' %>
+            <% end %>
+          </div>
+        </div>
+
+        <div class="panel panel-default">
+          <div class="panel-heading">
+            <h4><%= t('.time_sheets_overview')%></h4>
+          </div>
+          <div class="panel-body">
+            <% @user.years_and_months_of_existence.each do |month| %>
+                <p>
+                  <%= link_to l(Date.new(month.first, month.last, 1), format: :without_day) + ':', work_days_path(month: month.last, year: month.first) %>
+                  <% current_user.projects_for_month(month.first, month.last).each do |project| %>
+                      <%= link_to project.title + " ", work_days_path(month: month.last, year: month.first, project: project.id, user_id: current_user) %>
+                  <% end %>
+                </p>
+            <% end %>
+          </div>
+        </div>
+    <% end %>
```

Comment:
Entweder spackt die Anzeige gerade oder in den Zeilen 25 bis 61 sollte die Einrückung noch einmal geprüft werden.